pkg/chains/formats/slsa/v2/taskrun/taskrun.go

```diff
@@ -70,17 +70,32 @@ func GenerateAttestation(builderID string, payloadType config.PayloadType, tro *
 // which material the Task definition came from
 func invocation(tro *objects.TaskRunObject) slsa.ProvenanceInvocation {
 	i := slsa.ProvenanceInvocation{}
-	if p := tro.Status.Provenance; p != nil {
+	if p := tro.Status.Provenance; p != nil && p.ConfigSource != nil {
 		i.ConfigSource = slsa.ConfigSource{
 			URI:        p.ConfigSource.URI,
 			Digest:     p.ConfigSource.Digest,
 			EntryPoint: p.ConfigSource.EntryPoint,
 		}
 	}
 	i.Parameters = invocationParams(tro)
+	env := invocationEnv(tro)
+	if len(env) > 0 {
+		i.Environment = env
+	}
 	return i
 }
 
+// invocationEnv adds the tekton feature flags that were enabled
+// for the taskrun. In the future, we can populate versioning information
+// here as well.
+func invocationEnv(tro *objects.TaskRunObject) map[string]any {
+	var iEnv map[string]any = make(map[string]any)
+	if tro.Status.Provenance != nil && tro.Status.Provenance.FeatureFlags != nil {
+		iEnv["tekton-config-feature-flags"] = tro.Status.Provenance.FeatureFlags
```

Comment:
nit:

```suggestion
		iEnv["tekton-pipelines-feature-flags"] = tro.Status.Provenance.FeatureFlags
```

so there's no confusion about what component these are coming from.

Reply:
Agreed! Thanks for the suggestion.